remote-monitoring/single-vm-setup.sh

```diff
@@ -25,6 +25,36 @@ export PCS_IOTHUBREACT_HUB_PARTITIONS="${12}"
 export PCS_IOTHUBREACT_AZUREBLOB_ACCOUNT="${13}"
 export PCS_IOTHUBREACT_AZUREBLOB_KEY="${14}"
 
+# ========================================================================
+
+# Install Docker engine based on host name if it is not installed. if the
+# host name has .cn suffix, it is expected on Mooncake and mirror option
+# will be used because of slow network traffic always fails the installation.
+function install_docker() {
+    set +e
+    local host_name=$1
+
+    TEST=$(which docker)
+    if [[ ! -z "$TEST" ]]; then
+        return;
+    fi
+
+    if [[ $host_name =~ ..*cn ]]; then
```

Comment:
If the host name has .cn suffix,  dockerhub in China will be used to avoid slow network traffic failure.